src/Discord.Net.Rest/Entities/Interactions/RestInteraction.cs

```diff
@@ -297,6 +299,17 @@ async Task<IUserMessage> IDiscordInteraction.FollowupWithFileAsync(FileAttachmen
         /// <inheritdoc/>
         async Task<IUserMessage> IDiscordInteraction.FollowupWithFilesAsync(IEnumerable<FileAttachment> attachments, string text, Embed[] embeds, bool isTTS, bool ephemeral, AllowedMentions allowedMentions, MessageComponent components, Embed embed, RequestOptions options)
             => await FollowupWithFilesAsync(attachments, text, embeds, isTTS, ephemeral, allowedMentions, components, embed, options).ConfigureAwait(false);
+        /// <inheritdoc/>
+        Task IDiscordInteraction.RespondWithFilesAsync(IEnumerable<FileAttachment> attachments, string text, Embed[] embeds, bool isTTS, bool ephemeral, AllowedMentions allowedMentions, MessageComponent components, Embed embed, RequestOptions options) => throw new NotSupportedException("REST-Based interactions don't support files.");
+        /// <inheritdoc/>
+#if NETCOREAPP3_0_OR_GREATER != true
+        /// <inheritdoc/>
+        Task IDiscordInteraction.RespondWithFileAsync(Stream fileStream, string fileName, string text, Embed[] embeds, bool isTTS, bool ephemeral, AllowedMentions allowedMentions, MessageComponent components, Embed embed, RequestOptions options) => throw new NotSupportedException("REST-Based interactions don't support files.");
+        /// <inheritdoc/>
+        Task IDiscordInteraction.RespondWithFileAsync(string filePath, string fileName, string text, Embed[] embeds, bool isTTS, bool ephemeral, AllowedMentions allowedMentions, MessageComponent components, Embed embed, RequestOptions options) => throw new NotSupportedException("REST-Based interactions don't support files.");
+        /// <inheritdoc/>
+        Task IDiscordInteraction.RespondWithFileAsync(FileAttachment attachment, string text, Embed[] embeds, bool isTTS, bool ephemeral, AllowedMentions allowedMentions, MessageComponent components, Embed embed, RequestOptions options) => throw new NotSupportedException("REST-Based interactions don't support files.");
+#endif
```

Comment:
REST/webhook interactions do support sending files though. I tested it by manually responding a slash command with a file and it works.
I'm not sure how this could be implemented in the library.

Reply:
How did you return the file via rest?

Reply:
I'm using ASP.NET Core to respond the requests. I made a minimal controller that responds any valid interaction (except pings) with a file:

```cs
using System.Net.Http.Headers;
using System.Text;
using Discord.Rest;
using Microsoft.AspNetCore.Mvc;

namespace InteractionsBot.Controllers;

[ApiController]
[Route("api/[controller]")]
public class InteractionsController : ControllerBase
{
    private readonly ILogger<InteractionsController> _logger;
    private readonly DiscordRestClient _client;
    private readonly HttpClient _httpClient;
    private readonly IConfiguration _configuration;

    public InteractionsController(ILogger<InteractionsController> logger, DiscordRestClient client,
        HttpClient httpClient, IConfiguration configuration)
    {
        _logger = logger;
        _client = client;
        _httpClient = httpClient;
        _configuration = configuration;
    }

    [HttpPost]
    public async Task PostInteractionsAsync()
    {
        var signature = Request.Headers["X-Signature-Ed25519"];
        var timestamp = Request.Headers["X-Signature-Timestamp"];
        using var reader = new StreamReader(Request.Body);
        var body = await reader.ReadToEndAsync();

        if (!_client.IsValidHttpInteraction(_configuration["PublicKey"], signature, timestamp, body))
        {
            Response.StatusCode = 401;
            Response.ContentType = "application/json";
            await Response.WriteAsync("Invalid signature.");
            await Response.CompleteAsync();
            return;
        }

        var interaction = await _client.ParseHttpInteractionAsync(_configuration["PublicKey"], signature, timestamp, body);

        if (interaction is RestPingInteraction pingInteraction)
        {
            Response.StatusCode = 200;
            Response.ContentType = "application/json";
            await Response.WriteAsync(pingInteraction.AcknowledgePing());
            await Response.CompleteAsync();
            return;
        }

        var stream = await _httpClient.GetStreamAsync("https://cdn.discordapp.com/attachments/698610179444441168/919319545854963742/unknown.png");

        var content = new MultipartFormDataContent($"Upload----{DateTimeOffset.UtcNow}");

        string payload =
@"{
    ""type"":4,
    ""data"":
    {
        ""content"": ""Hello World"",
        ""attachments"":
        [{
            ""id"":0,
            ""filename"":""test.png""
        }]
    }
}";

        var stringContent = new StringContent(payload, Encoding.UTF8, "application/json");
        stringContent.Headers.ContentDisposition = new ContentDispositionHeaderValue("form-data") { Name = "\"payload_json\"" };
        content.Add(stringContent);

        var streamContent = new StreamContent(stream);
        streamContent.Headers.ContentType = new MediaTypeHeaderValue("image/png");
        streamContent.Headers.ContentDisposition = new ContentDispositionHeaderValue("form-data") { Name = "files[0]", FileName = "\"test.png\"" };
        content.Add(streamContent);

        Response.ContentLength = content.Headers.ContentLength;
        Response.ContentType = content.Headers.ContentType!.ToString();

        await content.CopyToAsync(Response.Body);
        await Response.CompleteAsync();
    }
}
```